src/controller/node.c

```diff
@@ -513,14 +513,7 @@ static int node_match_job_done(UNUSED sd_bus_message *m, UNUSED void *userdata,
         return 1;
 }
 
-static int node_match_heartbeat(UNUSED sd_bus_message *m, void *userdata, UNUSED sd_bus_error *error) {
-        Node *node = userdata;
-
-        int r = get_time_seconds(&node->last_seen);
-        if (r < 0) {
-                bc_log_errorf("Failed to get current time on heartbeat: %s", strerror(-r));
-                return 0;
-        }
+static int node_match_heartbeat(UNUSED sd_bus_message *m, UNUSED void *userdata, UNUSED sd_bus_error *error) {
         return 1;
```

Comment:
Isn't the method useless? If so can't we remove it?

Reply:
Its not needed since the Heartbeat is just a signal emitted by the agent - so no listener is required. We wanted to have such a signal to test the connectivity on an application layer and had only a log statement, initially. I think that doesn't make sense and without updating the timestamp there the heartbeat mechanism seems a bit pointless (since we don't check the connectivity).
Maybe we should keep the current state and not do that optimization? What do you think? @mkemel @mwperina